src/core/send-commercial-metrics.ts

```diff
@@ -187,12 +187,17 @@ function gatherMetricsOnPageUnload(): void {
 const listener = (e: Event): void => {
 	switch (e.type) {
 		case 'visibilitychange':
-			if (document.visibilityState === 'hidden') {
+			if (
+				document.visibilityState === 'hidden' &&
+				window.guardian.config.shouldSendCommercialMetrics === true
+			) {
 				gatherMetricsOnPageUnload();
 			}
 			return;
 		case 'pagehide':
-			gatherMetricsOnPageUnload();
+			if (window.guardian.config.shouldSendCommercialMetrics === true) {
```

Comment:
Could the if go around the switch statement?

Reply:
Good shout! Done